### PR TITLE
[Metabot] Initial e2e + unit tests

### PR DESCRIPTION
**Pull request by sloansparger**
Basic e2e and unit test coverage for the UI and some unit tests to pin down some basic logic around context and history.

### CODE REVIEW

File: e2e/support/helpers/e2e-metabot-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import type { StaticResponse } from "cypress/types/net-stubbing";
+
+import { popover } from "./e2e-ui-elements-helpers";
+
+export function assertChatVisibility(visiblility: "visible" | "not.visible") {
+  cy.findByTestId("metabot-chat").should(
+    visiblility === "visible" ? "exist" : "not.exist",
```
Comment (iethree) [inline e2e/support/helpers/e2e-metabot-helpers.ts:7]:
```suggestion
    visiblility === "visible" ? "be.visible" : "not.exist",
```

you want the more robust assertion of visibility

File: e2e/support/helpers/e2e-metabot-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+export function openMetabotViaNewMenu(assertVisibility = true) {
+  if (assertVisibility) {
+    assertChatVisibility("not.visible");
+  }
+
+  cy.findByLabelText("Navigation bar").within(() => {
+    cy.findByText("New").click();
+  });
+  popover().findByText("Metabot request").click();
```
Comment (iethree) [inline e2e/support/helpers/e2e-metabot-helpers.ts:43]: there's a helper for that!

```
newButton("Metabot request");
```

maybe makes this helper moot, WDYT?

File: e2e/test/scenarios/metabot/metabot.cy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    it("should not be avaiable in OSS", () => {
+      openMetabotViaShortcutKey(false);
+      assertChatVisibility("not.visible");
+      cy.findByLabelText("Navigation bar").within(() => {
+        cy.findByText("New").click();
+      });
+      popover().findByText("Metabot request").should("not.exist");
+      assertChatVisibility("not.visible");
+    });
```
Comment (iethree) [inline e2e/test/scenarios/metabot/metabot.cy.spec.ts:39]: 👏 nice job remembering to test this case

File: e2e/test/scenarios/metabot/metabot.cy.spec.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    it("should allow a user to send a message to the agent and handle successful or failed responses", () => {
+      openMetabotViaShortcutKey();
+      userMessages().should("not.exist");
+
+      mockMetabotResponse({
+        statusCode: 200,
+        delay: 100, // small delay to detect loading state
+        body: whoIsYourFavoriteResponse,
+      });
```
Comment (iethree) [inline e2e/test/scenarios/metabot/metabot.cy.spec.ts:64]: This is interesting, It makes sense for testing the UI in a deterministic way, I wonder have you read up on what is the nascent state of the art for testing non-deterministic LLM responses?

File: enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+  });
+}
+
+const chat = () => screen.findByTestId("metabot-chat");
+const input = () => screen.findByTestId("metabot-chat-input");
+const enterChatMessage = async (message: string, send = true) =>
+  userEvent.type(await input(), `${message}${send ? "{Enter}" : ""}`);
+const closeChatButton = () => screen.findByTestId("metabot-close-chat");
+
+// NOTE: for some reason the keyboard shortcuts won't work with tinykeys while testing, using redux for now...
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx:62]: 😬

File: enterprise/frontend/src/metabase-enterprise/metabot/state/actions.ts [hunk 6]
```diff
@@ -76,8 +76,13 @@ export const sendMessageRequest = createAsyncThunk(
     );
 
     if (result.error) {
+      console.error("Metabot request returned error: ", result.error);
       dispatch(clearUserMessages());
-      dispatch(addUserMessage(t`I can’t do that, unfortunately.`));
+      const message =
+        (result.error as any).status === 500
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/metabot/state/actions.ts:82]: maybe?

```suggestion
        (result.error as any).status >= 400
```

**Review by iethree — APPROVED** [review]

File: e2e/test/scenarios/metabot/metabot.cy.spec.ts (hunk 4 above)
Comment (sloansparger) [reply e2e/test/scenarios/metabot/metabot.cy.spec.ts:64]: I am getting up to speed in this space, but I think collectively we're not yet sure about some core design decisions which greatly impact how we should test things. John and I are about to explore changing the API significantly, so once these things settle that's the next natural problem to solve.

File: enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx (hunk 5 above)
Comment (sloansparger) [reply enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx:62]: I have some grey hairs from this. I'll bug you or someone tomorrow, maybe I was doing something stupid.

File: e2e/support/helpers/e2e-metabot-helpers.ts (hunk 2 above)
Comment (sloansparger) [reply e2e/support/helpers/e2e-metabot-helpers.ts:43]: I think the asserts are still nice, but I'll definitely take advantage of that helper.

**Pull request merged**